wemake_python_styleguide/visitors/ast/classes.py

```diff
@@ -118,28 +119,21 @@ def _is_correct_base_class(self, base_class: ast.AST) -> bool:
         return False
 
     def _check_getters_setters_methods(self, node: ast.ClassDef) -> None:
-        class_attributes, instance_attributes = classes.get_attributes(
-            node,
-            include_annotated=True,
-        )
-        flat_class_attributes = name_nodes.flat_variable_names(class_attributes)
-
-        attributes_stripped = {
-            class_attribute.lstrip(constants.UNUSED_PLACEHOLDER)
-            for class_attribute in flat_class_attributes
-        }.union({
-            instance.attr.lstrip(constants.UNUSED_PLACEHOLDER)
-            for instance in instance_attributes
-        })
-
-        for method in classes.find_getters_and_setters(node):
-            if method.name[classes.GETTER_LENGTH:] in attributes_stripped:
-                self.add_violation(
-                    oop.UnpythonicGetterSetterViolation(
-                        method,
-                        text=method.name,
-                    ),
-                )
+        getters_and_setters = set(filter(
+            lambda getter_setter: functions.is_method(
+                getattr(getter_setter, 'function_type', None),
+            ),
+            getters_setters.find_paired_getters_and_setters(node),
+        )).union(set(
```

Comment:
Can you please write two comments for both `.union` parts?

Reply:
Basically, we need to explain why do we have two `set`s here.

Reply:
Done